src/Poseidon/CLI/Genoconvert.hs

```diff
@@ -98,30 +98,40 @@ convertGenoTo outFormat onlyGeno outPath removeOld inPlinkPopMode outPlinkPopMod
                     let outConsumer = case outFormat of
-                            GenotypeFormatEigenstrat -> writeEigenstrat outG outS outI eigenstratIndEntries
-                            GenotypeFormatPlink -> writePlink outG outS outI (map (eigenstratInd2PlinkFam outPlinkPopMode) eigenstratIndEntries)
+                            "EIGENSTRAT" -> writeEigenstrat outG outS outI eigenstratIndEntries
+                            "PLINK"      -> writePlink outG outS outI (map (eigenstratInd2PlinkFam outPlinkPopMode) eigenstratIndEntries)
+                            _  -> liftIO . throwIO $ PoseidonGenericException ("Illegal outFormat " ++ outFormat ++ ". Only Outformats EIGENSTRAT or PLINK are allowed at the moment")
                     runEffect $ eigenstratProd >-> printSNPCopyProgress logA currentTime >-> outConsumer
                 ) (throwIO . PoseidonGenotypeExceptionForward errLength)
             logInfo "Done"
             -- overwrite genotype data field in POSEIDON.yml file
             unless (onlyGeno || isJust outPath) $ do
-                let genotypeData = GenotypeDataSpec outFormat outGeno Nothing outSnp Nothing outInd Nothing (snpSet . posPacGenotypeData $ pac)
+                gFileSpec <- case outFormat of
+                        "EIGENSTRAT" -> return $ GenotypeEigenstrat outGeno Nothing outSnp Nothing outInd Nothing
+                        "PLINK"      -> return $ GenotypePlink      outGeno Nothing outSnp Nothing outInd Nothing
+                        _  -> liftIO . throwIO $ PoseidonGenericException ("Illegal outFormat " ++ outFormat ++ ". Only Outformats EIGENSTRAT or PLINK are allowed at the moment")
+                let genotypeData = GenotypeDataSpec gFileSpec (genotypeSnpSet . posPacGenotypeData $ pac)
                     newPac = pac { posPacGenotypeData = genotypeData }
                 logInfo "Adjusting POSEIDON.yml..."
                 liftIO $ writePoseidonPackage newPac
             -- delete now replaced input genotype data
-            when removeOld $ liftIO $ mapM_ removeFile [
-                  posPacBaseDir pac </> genoFile (posPacGenotypeData pac)
-                , posPacBaseDir pac </> snpFile  (posPacGenotypeData pac)
-                , posPacBaseDir pac </> indFile  (posPacGenotypeData pac)
-                ]
+            let filesToDelete = case genotypeFileSpec . posPacGenotypeData $ pac of
+                    GenotypeEigenstrat g _ s _ i _ -> [g, s, i]
+                    GenotypePlink      g _ s _ i _ -> [g, s, i]
+                    GenotypeVCF        g _         -> [g]
+            when removeOld . liftIO . mapM_ (removeFile . (posPacBaseDir pac </>)) $ filesToDelete
   where
     checkFile :: FilePath -> PoseidonIO Bool
     checkFile fn = do
         fe <- liftIO $ doesFileExist fn
         when fe $ logWarning $ "File " ++ fn ++ " exists"
         return fe
+    getFormat :: GenotypeFileSpec -> String
```

Comment:
This looks like a function that will also be useful elsewhere. Maybe it should be defined where the type is defined.